projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/AuthenticationKey.java

```diff
@@ -0,0 +1,92 @@
+package org.batfish.datamodel;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Date;
+import org.batfish.common.util.ComparableStructure;
+
+public class AuthenticationKey extends ComparableStructure<String> {
+
+  public enum KeyOption {
+    BASIC,
+    ISIS_ENHANCED
+  }
+
+  private static final String PROP_ALGORITHM = "algorithm";
+
+  private static final String PROP_OPTION = "option";
+
+  private static final String PROP_SECRET = "secret";
+
+  private static final String PROP_START_TIME = "startTime";
+
+  private static final long serialVersionUID = 1L;
+
+  private BgpAuthenticationAlgorithm _algorithm;
+
+  private AuthenticationKey.KeyOption _option;
+
+  private String _secret;
+  
+  private Date _startTime;
```

Comment:
Shouldn't store timeStamp as a Java Date object? What should I use and what is the difference? Thanks